From db656be7d8a03f460519fd91231a267facf8a474 Mon Sep 17 00:00:00 2001
From: MerCuJerry <mercujerry@gmail.com>
Date: Fri, 21 Nov 2025 23:32:56 +0800
Subject: [PATCH] after

---
 fs/notify/fdinfo.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/fs/notify/fdinfo.c b/fs/notify/fdinfo.c
index 439799714002..9755856f1869 100644
--- a/fs/notify/fdinfo.c
+++ b/fs/notify/fdinfo.c
@@ -122,10 +122,11 @@ static void inotify_fdinfo(struct seq_file *m, struct fsnotify_mark *mark)
 			if (kern_path(dpath, 0, &path)) {
 				goto out_free_pathname;
 			}
-			seq_printf(m, "inotify wd:%x ino:%lx sdev:%x mask:%x ignored_mask:0 ",
-			   inode_mark->wd, path.dentry->d_inode->i_ino, path.dentry->d_inode->i_sb->s_dev,
-			   inotify_mark_user_mask(mark));
-			show_mark_fhandle(m, path.dentry->d_inode);
+			u32 mask = mark->mask & IN_ALL_EVENTS;
+			seq_printf(m, "inotify wd:%x ino:%lx sdev:%x mask:%x ignored_mask:%x ",
+				inode_mark->wd, inode->i_ino, inode->i_sb->s_dev,
+				mask, mark->ignored_mask);
+			show_mark_fhandle(m, inode);
 			seq_putc(m, '\n');
 			iput(inode);
 			path_put(&path);
-- 
2.52.0

